### ISSUE TITLE
Corrupted images error

### ISSUE THREAD

**Issue by Happinesseuh** [ultralytics/yolov5#3095]
I tried to train custom data, but when i launch the train i have this issue :

![Screenshot_20210510_130402](https://user-images.githubusercontent.com/57877506/117649871-52f30f80-b190-11eb-941f-041a47b239fa.png)

We can see that the images are corrupted, but is the good format (jpg) and the good path to the folder.

How can I solve it ?

**Comment by glenn-jocher** [comment]
@Happinesseuh good news 😃! This issue may be improved ✅ in PR #3103.  This won't solve your corrupted image problem but it should allow you to better understand the cause of the problem by logging to screen the cause, which seems not to be logging to screen with the default older print() statement in your case.
https://github.com/ultralytics/yolov5/blob/25f8ab835ef82a5664f6434934c7f40088117f65/utils/datasets.py#L491-L493

To receive this update you can:
- `git pull` from within your `yolov5/` directory
- `git clone https://github.com/ultralytics/yolov5` again
- Force-reload [PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/): `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- View our updated notebooks:  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by vrtompki** [comment]
@Happinesseuh I ran into this issue today, and it turned out to be a round off error after the labels were read and converted into Numpy arrays. Since I am working with "dense" small objects, the coordinates matched for the first 11 out of 17 digits so once converted to float32, the numbers were rounded to be the same 5 digit coords hence the corrupted/duplicate images error for me. So you could also look at your labels to see if that's potentially what happened in your case.

**Comment by kstisser** [comment]
Good afternoon, I am attempting to train using the [bird dataset ](http://www.vision.caltech.edu/visipedia/CUB-200.html), where I hand labeled the first 34 species using the [labelImg ](https://github.com/tzutalin/labelImg) tool. I am also getting the 'ignoring corrupt image/label', though the files exist. I am running [this yaml](https://github.com/kstisser/yolov5/blob/birdTraining/myData/birds.yaml), you can see in my fork with data/labels included. I am running the latest branch today with the following command:
python train.py --img 416 --batch 12 --epochs 50 --data ./myData/birds.yaml --weights ./myData/weights/yolov5x.pt
Here is a screenshot. I'd appreciate any feedback. Thanks!
![yolov5_corrupt_image](https://user-images.githubusercontent.com/11004183/144723262-8f3f1010-3d36-447e-8989-6e970c745371.png)
!

**Comment by kstisser** [comment]
In doing some debugging it looks like it was having a hard time finding the files within the train.txt and test.txt, as I had just stated the file names. However it needed reference to the local directory added in front './' to make it './filename.jpg' in case this helps anyone else. This issue is resolved. Cheers!

**Comment by yejinaCodes** [comment]
![error](https://user-images.githubusercontent.com/77040445/156349264-931a6439-2f14-4cf3-8126-4344235f5ae6.png)

this is my error when I try to train Yolov5s using my custom dataset and the yaml I created.
the warning signs that were printed before the error said "ignoring corrupt image/label: could not convert string to float:'0.7736...'". Then I get the above ValueError message. Could I know what my problem is?

**Comment by glenn-jocher** [comment]
@YejinKimHanyang your images have problems. Please review your dataset for corrupted images prior to training.

**Comment by gmt710** [comment]
maybe your images and labels‘ folder do not exist.

**Comment by sebasmos** [comment]
It is a formatting issue by the data side. [#3103](https://github.com/ultralytics/yolov5/pull/3103) just prints the error. Debugging like this can help you find the error:
`[~/dataset.py]`
1. Check that the label contents on `cache_path` match the real files content.
2. See if `np.load(cache_path, allow_pickle=True).item()` actually works, if not then change the way data is being stored: wrong format etc.

**Comment by yejinaCodes** [comment]
> @YejinKimHanyang your images have problems. Please review your dataset for corrupted images prior to training.

Thank you! I solved it! My dataset was the problem. It got corrupted while I was changing it to COCO data format.

**Comment by nihanaltaytas** [comment]
![eror train](https://user-images.githubusercontent.com/118836059/207822525-b889e03f-b576-4f03-88f3-5dd885573429.JPG)
I have this similar problem.

**Comment by glenn-jocher** [comment]
@nihanaltaytas 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. VisDrone automatically downloads and starts training without any issues for me in my test just now, I am not able to reproduce any problems with it.

<img width="1280" alt="Screenshot 2022-12-17 at 12 13 31" src="https://user-images.githubusercontent.com/26833433/208239003-03b8889f-1c84-487e-b18d-acec7ec50d29.png">

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by anushkjd** [comment]
![Screenshot (96)](https://github.com/ultralytics/yolov5/assets/139153170/db4ba942-321e-4956-b631-3224095ffecf)
i have this problem

**Comment by anushkjd** [comment]
![Screenshot (132)](https://github.com/ultralytics/yolov5/assets/139153170/91cfabf0-9295-476f-b60a-0ef4527895e2)
i have this problem

**Comment by glenn-jocher** [comment]
@anushkjd this issue can occur when there is a mismatch between the number of classes specified in the YAML file and the actual number of classes in your dataset. Make sure that you have correctly defined the number of classes in the "nc" field of the YAML file.

If the number of classes is correct, please ensure that your dataset annotations and images are properly formatted and aligned. Double-check that the file paths in the annotations match the actual image locations. Additionally, verify that all images and annotations are valid and can be opened and parsed correctly.

If the problem persists, please consider providing a minimal, reproducible example along with your code and dataset to help us further investigate the issue.

Thank you for your understanding, and we will do our best to assist you with this problem.
